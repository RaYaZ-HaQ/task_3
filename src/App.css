.flex-container {
  display: flex;
}

.flex-container-column {
  display: flex;
  flex-direction: column;
}

.items-center {
  align-items: center;
}

.flex-1 {
  flex: 1 1 0;
}

.box {
  background: #ffffff;
}

#box-1 {
  display: none;
}

.spacing-20 {
  gap: 1.25rem;
}

.padding-20 {
  padding: 1.25rem;
}

.border-black {
  border: 1px solid black;
  border-radius: 1rem;
}

.image-container {
  margin-bottom: auto; /* Prevent container height changing due to box-3 height changes */
  overflow: hidden;
}

.image-container img {
  max-width: 100%;
}

@media screen and (min-width: 800px) {
  #box-1 {
    display: block;
  }

  #box-3 {
    max-width: 25%;
    word-break: break-all;
  }
}
