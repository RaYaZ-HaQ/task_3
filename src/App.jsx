import "./App.css";
import Form from "./components/GUIDForm";

function App() {
  return (
    <main className="flex-container spacing-20 padding-20">
      <div className="flex-container-column flex-1 spacing-20">
        <div className="box border-black padding-20">
          <Form />
        </div>
        <div id="box-3" className="box border-black padding-20 max-w-1/4">
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit.
            Reprehenderit nisi nobis labore repudiandae dolores incidunt
            voluptatibus architecto, quos dignissimos, qui fugiat magnam. Dicta,
            iure optio aspernatur dignissimos perspiciatis quo voluptatem!
          </p>
        </div>
      </div>
      <div id="box-1" className="box image-container flex-1 border-black">
        <img src="/wallhaven-o52y1l.jpg" alt="Test image" />
      </div>
    </main>
  );
}

export default App;
