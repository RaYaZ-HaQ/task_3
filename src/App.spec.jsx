import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

describe("Application tests", () => {
  it("should match snapshot", () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });

  it("should container 3 boxes", () => {
    render(<App />);

    const boxes = document.getElementsByClassName("box");
    expect(boxes).toHaveLength(3);
  });
});
