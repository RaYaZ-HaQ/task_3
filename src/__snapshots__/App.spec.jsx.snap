// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application tests should match snapshot 1`] = `
<div>
  <main
    class="flex-container spacing-20 padding-20"
  >
    <div
      class="flex-container-column flex-1 spacing-20"
    >
      <div
        class="box border-black padding-20"
        id="box-2"
      >
        <form
          class="flex-container-column spacing-20"
        >
          <span>
            <label>
              Enter a valid GUID here.
            </label>
          </span>
          <input
            id="guid"
            name="guid"
            type="text"
          />
          <button
            class="button flex-container items-center border-black"
          >
            <span>
              Submit
            </span>
          </button>
        </form>
      </div>
      <div
        class="box border-black padding-20 max-w-1/4"
        id="box-3"
      >
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Reprehenderit nisi nobis labore repudiandae dolores incidunt voluptatibus architecto, quos dignissimos, qui fugiat magnam. Dicta, iure optio aspernatur dignissimos perspiciatis quo voluptatem!
        </p>
      </div>
    </div>
    <div
      class="box image-container flex-1 border-black"
      id="box-1"
    >
      <img
        alt="Test image"
        src="/wallhaven-o52y1l.jpg"
      />
    </div>
  </main>
</div>
`;
